### PR TITLE
cmd/contour: unify client creation

### PR DESCRIPTION
**Pull request by davecheney**
Updates #403

Before I can add ingress status support to cmd/contour I need to clean
up the serve method. This change unifies the creation of the various
clients to a single method and improves the naming of the various return
values -- they are infomer _factories_ not informers.

Signed-off-by: Dave Cheney <dave@cheney.net>

### CODE REVIEW

File: cmd/contour/clients.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	"k8s.io/client-go/dynamic/dynamicinformer"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	coordinationv1 "k8s.io/client-go/kubernetes/typed/coordination/v1"
+)
+
+// kubernetesClients holds the various API clients required by Contour.
+type kubernetesClients struct {
```
Comment (jpeach) [inline cmd/contour/clients.go]: Can we put this in the `internal.k8s` package along with the other generic Kubernetes helpers?

Comment (davecheney) [reply cmd/contour/clients.go]: Sure thing. Please hold

File: internal/k8s/clients.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+// NewDynamicInformerFactory returns a new DynamicSharedInformerFactory for
+// use with any registered Kubernetes API type.
+func (c *Clients) NewDynamicInformerFactory() dynamicinformer.DynamicSharedInformerFactory {
+	return dynamicinformer.NewDynamicSharedInformerFactory(c.dynamic, resyncInterval)
+}
+
+// ClientSet returns the Kubernetes Core v1 ClientSet.
+func (c *Clients) ClientSet() *kubernetes.Clientset {
```
Comment (davecheney) [inline internal/k8s/clients.go]: These getters are unpleasant, I'll try to clean them up in a future PR.

Comment (jpeach) [reply internal/k8s/clients.go]: Yeh, I'd be inclined to just make the struct members public, but this is fine too.

**Pull request merged**